### DISCUSSION TITLE
MVVM VS Another Simple Pattern

### DISCUSSION THREAD

**Discussion by alizaenazet** [jerrytjahyono/SelfFit#3] [General]
```
disclamer: pertanyaan dan opini berdasarkan latarbelakang pengetahuan SwiftUi yang terbatas dan tidak memiliki pengalaman mencoba pattern apapun sebelumnya.
```

Beberapa artikel menui pro dan kontra terhadap arsitektur populer pada Swiftui app yaitu `MVVM` dimana banyak yang akhirnya menyarankan Pattern yang cenderung baru alias kurang populer seperti TCA, MV bahkan MVC.

dari beberapa artikel yang telah aku baca mendapati beberapa kesimpulan sebagai berikut :

### MVVM 
Populer dan dapat lebih mudah mendapati resource belajar di internet, namun ketika membaca beberapa artikel MVVM cenderung lebih rumit

### Pattern lainnya
cenderung dikatakan lebih sederhana dan lebih efektif dikarenakan tidak melakukan abuse terhadap kemampuan SwiftUi by default.

pada kasus ini dimana anggota kelompok belum perna belajar arsitektural pattern sebelumnya sehingga ini akan menjadi awal pemahan penerapan pattern pada swiftui seluruh anggota, disisi lain waktu development cukup terbatas sehingga perlu pertimbangkan selain untuk digunakan tetapi untuk pelajari.

ada beberapa temuan pattern yang baru : 
- TCA
- MV
- MVC

**Apakah sebagai awal belajar pattern untuk langsung belajar pattern TCA dan MV**

**Comment by alizaenazet** [comment]
Apakah sebaiknya belajar pattern `MV` saja ? sebab cenderung memiliki learning curve yang lebih cocok untuk pemula.

https://medium.com/better-programming/swiftui-architecture-a-complete-guide-to-mv-pattern-approach-5f411eaaaf9e

**Reply by TimothyKristanto** [reply]
Nice research, boleh banget pakek MV, pakek yang sekiranya bisa cocok dengan kamu dan tim mu yah, supaya bisa kolaborasinya lebih enak

**Reply by alizaenazet** [reply]
oke nice ko, sip sip 🫡 👍
